app/lib/verse8/credit.ts

```diff
@@ -0,0 +1,73 @@
+export async function getUserCredit(
+  endpoint: string,
+  userUid: string,
+  creditCredentials: { clientId: string; clientSecret: string },
+): Promise<bigint> {
+  try {
+    const response = await fetch(endpoint + `/v1/users/${userUid}`, {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-client-id': creditCredentials.clientId,
+        'x-client-secret': creditCredentials.clientSecret,
+      },
+    });
+
+    if (response.status === 404) {
+      return 0n;
+    }
+
+    if (!response.ok) {
+      throw new Error('Failed to get user credit');
+    }
+
+    const { credits } = (await response.json()) as {
+      credits: string;
+    };
+
+    return BigInt(credits);
+  } catch (error) {
+    console.error('Failed to get user credit', error);
+    throw new Error('Failed to get user credit');
+  }
+}
+
+export async function consumeUserCredit(
+  endpoint: string,
+  userUid: string,
+  creditCredentials: { clientId: string; clientSecret: string },
+  inputTokens: string,
+  outputTokens: string,
```

Comment:
Maybe knit-pick, but is there any reason to treat these values as `string`, instead of `number`? (e.g., token might have decimal points)

Reply:
I just quickly implemented it to match the API usage example, but it’d be better to use a number. Thanks!